src/negotiation-status-manager.ts

```diff
@@ -0,0 +1,149 @@
+  constructor(url : string) {
+    super();
+    if (url)
+      this._backendWs = new WebSocket(url);
+  }
+
+  allConflicts(): Record<number, NegotiationConflict> {
+    return this._conflicts;
+  }
+
+  startSubscription() {
+    if (this._backendWs) {
+      this._backendWs.send(
+        JSON.stringify({request: "negotiation_update_subscribe"}));
+
+      //only recv negotiation status messages
+      this._backendWs.onmessage = (event) => {
+        const msg = JSON.parse(event.data);
+        if (msg["type"] === "negotiation_status")
+        {
+          const conflictVersion : number = msg["conflict_version"];
+          const conflictVersionStr = conflictVersion.toString()
+      
+          let conflict = this._conflicts[conflictVersionStr];
+          if (conflict === undefined)
+          {
+            conflict = new NegotiationConflict();
+      
+            const participantId : number = msg["participant_id"]      
+            conflict.participantIdsToNames[participantId] = msg["participant_name"];
+      
+            this._conflicts[conflictVersionStr] = conflict;
+          }
+      
+          const id : number = msg["participant_id"];
+          const idStr = id.toString();
+          conflict.participantIdsToNames[idStr] = msg["participant_name"];
+      
+          let statusData = conflict.participantIdsToStatus[idStr];
+          let status : NegotiationStatus;
+          if (statusData === undefined)
+          {
+            statusData = new NegotiationStatusData();
+            conflict.participantIdsToStatus[idStr] = statusData;
+            status = statusData.base; 
+          }
+          else
+          {
+            const seq : number[] = msg["sequence"];
+            if (seq.length === 1)
+              status = statusData.base;
+            else
+            {
+              statusData.hasTerminal = true;
+              status = statusData.terminal;
+            }
+          }
+      
+          status.defunct = msg["defunct"];
+          status.rejected = msg["rejected"];
+          status.forfeited = msg["forfeited"];
+          status.sequence = msg["sequence"];
+      
+          this.emit('updated');
+        }
+        else if (msg["type"] === "negotiation_conclusion")
+        {
+          const conflictVersion : number = msg["conflict_version"];
+          const conflict = this._conflicts[conflictVersion.toString()];
+      
+          if (conflict === undefined)
+          {
+            console.warn('Undefined conflict version ' + conflictVersion + ', ignoring...');
+            return;
+          }
+      
+          if (msg["resolved"] === true)
+            conflict.resolved = ResolveState.RESOLVED;
+          else
+            conflict.resolved = ResolveState.FAILED;
+          
+          this.removeOldConflicts();
+        }
+      };
+    }
+  }
+
+  removeOldConflicts() {
+    const retain_count = 99; // increase this value for testing
```

Comment:
style fix 

> Now if we refresh the page all the conflicts are wiped out, right?
yes, that's correct.

> Not for his PR but in the future, do you think we should save the conflicts in the browser localStorage?
The only problem with that is if you restart the gazebo simulation conflicts will be retained from the previous simulation

Reply:
storing it in localStorage would also miss any negotiations that happens while the page is not loaded. I think this would need a service to query the list of negotiations available.

Reply:
> The only problem with that is if you restart the gazebo simulation conflicts will be retained from the previous simulation

That's a good point. 




> storing it in localStorage would also miss any negotiations that happens while the page is not loaded. I think this would need a service to query the list of negotiations available.

Oh! you're right, having a service to query the list of negotiations available sounds like a good solution.